Index: linux-6.1.89/arch/arm/boot/dts/stm32mp135f-dk.dts
===================================================================
--- linux-6.1.89.orig/arch/arm/boot/dts/stm32mp135f-dk.dts
+++ linux-6.1.89/arch/arm/boot/dts/stm32mp135f-dk.dts
@@ -494,6 +494,7 @@
 	pinctrl-1 = <&sdmmc2_b4_od_pins_a &sdmmc2_clk_pins_a>;
 	pinctrl-2 = <&sdmmc2_b4_sleep_pins_a>;
 	non-removable;
+	cap-sdio-irq;
 	st,neg-edge;
 	bus-width = <4>;
 	vmmc-supply = <&v3v3_ao>;
@@ -505,9 +506,6 @@
 	brcmf: bcrmf@1 {
 		reg = <1>;
 		compatible = "brcm,bcm4329-fmac";
-		interrupt-parent = <&gpiof>;
-		interrupts = <0 IRQ_TYPE_LEVEL_HIGH>; /* WL_HOST_WAKE */
-		interrupt-names = "host-wake";
 	};
 };
 
Index: linux-6.1.89/arch/arm/boot/dts/stm32mp157f-dk2.dts
===================================================================
--- linux-6.1.89.orig/arch/arm/boot/dts/stm32mp157f-dk2.dts
+++ linux-6.1.89/arch/arm/boot/dts/stm32mp157f-dk2.dts
@@ -127,6 +127,7 @@
 	pinctrl-1 = <&sdmmc2_b4_od_pins_a>;
 	pinctrl-2 = <&sdmmc2_b4_sleep_pins_a>;
 	non-removable;
+	cap-sdio-irq;
 	st,neg-edge;
 	bus-width = <4>;
 	vmmc-supply = <&v3v3>;
@@ -138,9 +139,6 @@
 	brcmf: bcrmf@1 {
 		reg = <1>;
 		compatible = "brcm,bcm4329-fmac";
-		interrupt-parent = <&gpiod>;
-		interrupts = <0 IRQ_TYPE_LEVEL_HIGH>; /* WL_HOST_WAKE */
-		interrupt-names = "host-wake";
 	};
 };
 
