Index: linux-6.1.89/drivers/rtc/rtc-stm32.c
===================================================================
--- linux-6.1.89.orig/drivers/rtc/rtc-stm32.c
+++ linux-6.1.89/drivers/rtc/rtc-stm32.c
@@ -1213,7 +1213,7 @@ static int stm32_rtc_resume(struct devic
 }
 
 static const struct dev_pm_ops stm32_rtc_pm_ops = {
-	SET_NOIRQ_SYSTEM_SLEEP_PM_OPS(stm32_rtc_suspend, stm32_rtc_resume)
+	NOIRQ_SYSTEM_SLEEP_PM_OPS(stm32_rtc_suspend, stm32_rtc_resume)
 };
 
 static struct platform_driver stm32_rtc_driver = {
@@ -1221,7 +1221,7 @@ static struct platform_driver stm32_rtc_
 	.remove		= stm32_rtc_remove,
 	.driver		= {
 		.name	= DRIVER_NAME,
-		.pm	= &stm32_rtc_pm_ops,
+		.pm	= pm_sleep_ptr(&stm32_rtc_pm_ops),
 		.of_match_table = stm32_rtc_of_match,
 	},
 };
