Index: linux-6.1.89/drivers/usb/host/Kconfig
===================================================================
--- linux-6.1.89.orig/drivers/usb/host/Kconfig
+++ linux-6.1.89/drivers/usb/host/Kconfig
@@ -128,7 +128,7 @@ config USB_BRCMSTB
 
 config USB_STM32_USBH
 	tristate "STM32 USB2 Host Glue support"
-	depends on ARCH_STM32 || COMPILE_TEST
+	depends on (ARCH_STM32 && CONFIG_ARM64) || COMPILE_TEST
 	select USB_OHCI_HCD_PLATFORM if USB_OHCI_HCD
 	select USB_EHCI_HCD_PLATFORM if USB_EHCI_HCD
 	help
